sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RequestOptions.java

```diff
@@ -452,4 +450,32 @@ public Duration getThresholdForDiagnosticsOnTracer() {
     public void setThresholdForDiagnosticsOnTracer(Duration thresholdForDiagnosticsOnTracer) {
         this.thresholdForDiagnosticsOnTracer = thresholdForDiagnosticsOnTracer;
     }
+
+    /**
+     * Gets the priority level of the request.
+     *
+     * When Priority Based Throttling is enabled, once the user has exhausted their provisioned throughput,
+     * low priority requests are throttled before high priority requests start getting throttled.
+     *
+     * Default PriorityLevel for each request is treated as High. It can be explicitly set to Low for some requests.
+     *
+     * @return enum representing priority level
+     */
+    public PriorityLevel getPriorityLevel() {
```

Comment:
Like mentioned in the discussion - I don't want this to be a competing feature for Java SDK customers. Instead priority level should be a configuration option on the ThroughputControlGroup. That way customers have a single way to deal with it - and can use the additinal features priority based throttling provides there. So, the public API changes would be around throughput control - not request options.